webapp/api/github.py

```diff
@@ -0,0 +1,143 @@
+import os
+import flask
+from urllib.parse import urlencode
+from webapp import api
+from requests.exceptions import RequestException
+
+GITHUB_CLIENT_ID = os.getenv("GITHUB_CLIENT_ID")
+GITHUB_CLIENT_SECRET = os.getenv("GITHUB_CLIENT_SECRET")
+
+
+class GithubAPIError(RequestException):
+    pass
+
+
+class GithubAPI:
+    """
+    Provides authentication for GitHub users. Helper methods are also provided
+    for checking organization access and getting user data from the Github API.
+    """
+
+    def __init__(self):
+        self.client_id = GITHUB_CLIENT_ID
+        self.client_secret = GITHUB_CLIENT_SECRET
```

Comment:
I think ideally it should take at least the client_ info and the session as arguments. The client stuff should be the client's responsibility to supply, so I think that should be passed through in oauth_views.py etc., and if you're not going to allow the user to override the base_url etc. (a perfectly valid choice) then they should be constants on the object:

``` python3
    BASE_URL = "https://api.github.com/"
    AUTH_URL = "https://github.com/login/oauth/"

    def __init__(
        self,
        client_id,
        client_secret,
        session=api.requests.Session()
    ):
        self.client_id = client_id
        self.client_secret = client_secret
        self.session = session
        self.session.headers["Accept"] = "application/json"
```

Reply:
Okay, I did it!